src/keylessBackup/KeylessBackupProgress.tsx

```diff
@@ -295,70 +323,127 @@ function Setup({ origin }: { origin: KeylessBackupOrigin }) {
 
   switch (keylessBackupStatus) {
     case KeylessBackupStatus.InProgress: {
-      return renderInProgressState(t('keylessBackupStatus.setup.inProgress.title'))
+      return (
+        <SafeAreaView style={styles.safeAreaView}>
+          {isOnboarding && (
+            <CustomHeader
+              title={
+                <HeaderTitleWithSubtitle
+                  title={t('keylessBackupStatus.setup.inProgress.title')}
+                  subTitle={t('registrationSteps', { step, totalSteps })}
+                />
+              }
+              style={styles.header}
+            />
+          )}
+          <ScrollView contentContainerStyle={styles.bodyContainer}>
+            <View style={iconMarginTop}>
+              <GreenLoadingSpinner />
+            </View>
+            {!isOnboarding && (
+              <Text style={styles.title}>{t('keylessBackupStatus.setup.inProgress.title')}</Text>
+            )}
+          </ScrollView>
+        </SafeAreaView>
+      )
     }
     case KeylessBackupStatus.Completed: {
       return (
-        <SafeAreaView style={styles.container}>
-          <View style={styles.finishedContainer}>
-            <View style={styles.iconContainer}>
+        <SafeAreaView style={styles.safeAreaView}>
+          {isOnboarding && (
```

Comment:
this can probably be made more DRY since the same safe area view / header wrapper is duplicated in every case. Could be in a follow up

Reply:
Fixed in 30e1497cf0936fecd68252c2e11295e4d74734da!